pandas/core/arrays/base.py

```diff
@@ -515,9 +517,17 @@ def nbytes(self) -> int:
     # Additional Methods
     # ------------------------------------------------------------------------
 
-    def astype(self, dtype, copy=True):
+    @overload
+    def astype(self, dtype: NpDtype, copy: bool = ...) -> np.ndarray:
```

Comment:
could this use npt.DtypeLike?

Reply:
> could this use npt.DtypeLike?

No.  The following code is fine with mypy
```
import numpy.typing as npt

class Foo:
    pass

def myfun(x: npt.DTypeLike):
    pass

myfun(type(Foo))
myfun("abcd")
myfun(str)
myfun(object)
myfun("str")
```
That's because `npt.DTypeLike` accepts any type or any str.  We want to separate out the `ExtensionDtype` that are passed as strings.

Reply:
Never mind.  Was able to get this to work by making `astype()` accept `npt.DTypeLike`.  Created a new type `AstypeArg` to mean `ExtensionDtype | npt.DTypeLike` so that I don't have to change lots of other parts where `NpDtype` is used.